Content.Server/Antag/AntagSelectionSystem.cs

```diff
@@ -0,0 +1,154 @@
+        var playerList = new List<IPlayerSession>();
+        var prefList = new List<IPlayerSession>();
+        chosen = new List<EntityUid>();
+        foreach (var player in allPlayers)
+        {
+            var mind = player.GetMind();
+            if (includeHeads == false)
+            {
+                if (!player.Data.ContentData()?.Mind?.AllRoles.All(role => role is not Job { CanBeAntag: false }) ?? false)
+                    continue;
+            }
+
+            if (player.AttachedEntity == null || HasComp<HumanoidAppearanceComponent>(player.AttachedEntity))
+                playerList.Add(player);
+            else
+                continue;
+
+            var pref = (HumanoidCharacterProfile) _prefs.GetPreferences(player.UserId).SelectedCharacter;
+            if (pref.AntagPreferences.Contains(antagPrototype))
+                prefList.Add(player);
+        }
+        if (playerList.Count == 0)
+            return;
+
+        var antags = Math.Clamp(allPlayers.Count / antagsPerPlayer, 1, maxAntags);
+        for (var antag = 0; antag < antags; antag++)
+        {
+            IPlayerSession chosenPlayer;
+            if (prefList.Count == 0)
+            {
+                if (playerList.Count == 0)
+                {
+                    break;
+                }
+                chosenPlayer = _random.PickAndTake(playerList);
+            }
+            else
+            {
+                chosenPlayer = _random.PickAndTake(prefList);
+                playerList.Remove(chosenPlayer);
+            }
+            var mind = chosenPlayer.Data.ContentData()?.Mind;
+            if (mind != null && mind.OwnedEntity != null)
+            {
+                chosen.Add((EntityUid) mind.OwnedEntity);
+                _audioSystem.PlayGlobal(antagSound, mind.OwnedEntity.Value);
+                if (mind.Session != null)
+                {
+                    var message = Loc.GetString(antagGreeting);
+                    var wrappedMessage = Loc.GetString("chat-manager-server-wrap-message", ("message", message));
+                    _chatManager.ChatMessageToOne(Shared.Chat.ChatChannel.Server, message, wrappedMessage, default, false, mind.Session.ConnectedClient, Color.FromHex(greetingColor));
+                }
+            }
+        }
+    }
+    /// <summary>
+    /// Will attempt to spawn an item inside of a persons bag. (Will be stuck to them if it can't, might just delete item if people care)
+    /// </summary>
+    /// <param name="antag">The entity that you want to spawn an item on</param>
+    /// <param name="item">The prototype ID that you want to spawn in the bag.</param>
+
+    public void GiveAntagBagGear(EntityUid antag, string item)
+    {
+        if (_inventory.TryGetSlotContainer(antag, "back", out var containerSlot, out var slotDefinition))
+        {
+            var bag = containerSlot.ContainedEntity;
+            var itemToSpawn = Spawn(item, new EntityCoordinates(antag, Vector2.Zero));
```

Comment:
if there is no bag wont this parent the flash to the headrev, instantly outing them on right click

Reply:
That's right that's what it says in the comment but since this is on round start they should have a bag on anyway, it's really only a problem if you admin command. I'll probably just remove it if there is no space for admin command though.

Reply:
it just deletes it lmao

Reply:
resolve now that it check pocket too